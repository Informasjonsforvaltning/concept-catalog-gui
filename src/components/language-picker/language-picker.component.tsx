import React from 'react';
import cx from 'classnames';
import { Button } from 'reactstrap';

import { localization } from '../../lib/localization';
import './language-picker.styles.scss';
import { Language } from '../../domain/Language';

interface Props {
  languages: Language[];
  toggleInputLanguage: Function;
}

export const LanguagePicker = ({ languages, toggleInputLanguage }: Props): JSX.Element => {
  const shouldDisableLanguage = (code): boolean => {
    const selectedLanguages = languages.filter(({ selected }) => selected);
    return selectedLanguages.length === 1 && selectedLanguages[0].code === code;
  };

  return (
    <div className="language-picker">
      <p>{`${localization.langChoose}:`}</p>
      <div className="language-button-group">
        {languages.map(({ code, title, selected }) => (
          <Button
            key={code}
            className={cx('fdk-button border-0', {
              'fdk-bg-color-primary-lighter': !selected,
              'fdk-color-link-darker': !selected,
              'no-shadow': selected
            })}
            color={shouldDisableLanguage(code) ? 'secondary' : 'primary'}
            disabled={shouldDisableLanguage(code)}
            onClick={() => toggleInputLanguage(code)}
          >
            {selected && <img src="./img/icon-checked-white-sm.svg" alt="icon" />}
            {title}
          </Button>
        ))}
      </div>
    </div>
  );
};
