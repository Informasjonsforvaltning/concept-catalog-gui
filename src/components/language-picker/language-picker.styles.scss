.language-picker {
  margin-bottom: 40px;

  p {
    margin-bottom: 10px;
    font-size: 110%;
  }

  .language-button-group {
    display: flex;

    :nth-of-type(n + 2) {
      margin-left: 10px;
    }

    button {
      display: flex;
      justify-content: center;
      align-items: center;

      img {
        height: 18px;
        width: 18px;
        margin-right: 5px;
      }

      &.no-shadow {
        box-shadow: none;
      }
    }
  }
}
