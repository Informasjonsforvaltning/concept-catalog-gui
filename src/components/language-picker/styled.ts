import styled from 'styled-components';

import { theme } from '@fellesdatakatalog/theme';

const LanguagePicker = styled.div`
  display: flex;
`;

const Label = styled.label`
  margin-right: ${theme.spacing('S12')};

  & > span {
    margin-left: ${theme.spacing('S6')};
  }
`;

export default { LanguagePicker, Label };
