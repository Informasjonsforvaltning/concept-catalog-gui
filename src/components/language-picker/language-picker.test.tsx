import React from 'react';
import { shallow } from 'enzyme';
import { LanguagePicker } from './language-picker.component';

let defaultProps;
let wrapper;

beforeEach(() => {
  defaultProps = {
    languages: [],
    toggleInputLanguage: () => {}
  };
  wrapper = shallow(<LanguagePicker {...defaultProps} />);
});

test('should render LanguagePicker correctly', () => {
  expect(wrapper).toMatchSnapshot();
});
