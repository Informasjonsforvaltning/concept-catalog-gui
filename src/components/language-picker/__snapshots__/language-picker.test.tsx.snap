// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LanguagePicker correctly 1`] = `
<div
  className="language-picker"
>
  <p>
    Velg spr√•k:
  </p>
  <div
    className="language-button-group"
  />
</div>
`;
